Makefile

```
@@ -123,26 +123,35 @@ test-e2e: ## Run e2e tests with capi e2e testing framework
 GINKGO_NOCOLOR ?= false
 ARTIFACTS ?= $(ROOT_DIR)/_artifacts
 E2E_CONF_FILE ?= $(ROOT_DIR)/test/e2e/config/e2e_conf.yaml
-E2E_TEMPLATES_DIR ?= $(ROOT_DIR)/templates/test
 E2E_ENVSUBST_DIR ?= $(ROOT_DIR)/test/e2e/_out
 E2E_CONF_FILE_ENVSUBST ?= $(E2E_ENVSUBST_DIR)/$(notdir $(E2E_CONF_FILE))
 E2E_CONTAINERS ?= quay.io/metal3-io/cluster-api-provider-metal3 quay.io/metal3-io/baremetal-operator quay.io/metal3-io/ip-address-manager
 
 SKIP_CLEANUP ?= false
 SKIP_CREATE_MGMT_CLUSTER ?= true
 
-$(E2E_ENVSUBST_DIR)/%.yaml: $(E2E_TEMPLATES_DIR)/%.yaml
-	mkdir -p $(E2E_ENVSUBST_DIR)
-	$(ENVSUBST) < $^ > $@
-
-## Processes all files from templates/test, as well as e2e_conf file
+## Processes e2e_conf file
 .PHONY: e2e-substitutions
-e2e-substitutions: $(ENVSUBST) $(subst $(E2E_TEMPLATES_DIR),$(E2E_ENVSUBST_DIR),$(wildcard $(E2E_TEMPLATES_DIR)/*.yaml))
+e2e-substitutions: $(ENVSUBST)
+	mkdir -p $(E2E_ENVSUBST_DIR)
 	$(ENVSUBST) < $(E2E_CONF_FILE) > $(E2E_CONF_FILE_ENVSUBST)
 
+## --------------------------------------
+## Templates
+## --------------------------------------
+E2E_TEMPLATES_DIR ?= $(ROOT_DIR)/test/e2e/data/infrastructure-metal3
+.PHONY: cluster-templates
+cluster-templates: $(KUSTOMIZE) ## Generate cluster templates
+	$(KUSTOMIZE) build $(E2E_TEMPLATES_DIR)/cluster-template-ubuntu --load-restrictor LoadRestrictionsNone > $(E2E_TEMPLATES_DIR)/cluster-template-ubuntu.yaml
+	$(KUSTOMIZE) build $(E2E_TEMPLATES_DIR)/cluster-template-centos --load-restrictor LoadRestrictionsNone > $(E2E_TEMPLATES_DIR)/cluster-template-centos.yaml
```

We should not need to change the load restrictor. We can instead change the kustomization (see other comments).
```suggestion
	$(KUSTOMIZE) build $(E2E_TEMPLATES_DIR)/cluster-template-ubuntu > $(E2E_TEMPLATES_DIR)/cluster-template-ubuntu.yaml
	$(KUSTOMIZE) build $(E2E_TEMPLATES_DIR)/cluster-template-centos > $(E2E_TEMPLATES_DIR)/cluster-template-centos.yaml
```

---

What do you mean by changing the kustomization

---

It is the [other comment](https://github.com/metal3-io/cluster-api-provider-metal3/pull/537#discussion_r814644367)